### PR TITLE
PODAUTO-199: Updates for 4.17

### PR DESCRIPTION
**Pull request by maxcao13**
Updates for 4.17

* Update deps
  * edit (update to 4.17/0.30.2) and run `hack/update-vendor.sh`
* Update code and build to match updated deps
 * Update go version to go 1.22 (go.mod, Dockerfile, images/ci/Dockerfile, Makefile) 
 * Switched from old images to updated ci tooling images `registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.22-openshift-4.17` in go-fmt, go-vet, Makefile, etc...
 * Rev version from 4.16 -> 4.17: `sed -i 's/4.16/4.17/g' $(git grep -l 4.16 manifests/) images/ci/bundle.Dockerfile hack/manifest-diff-upstream.sh hack/e2e.sh Makefile`
 * Verify that Dockerfile.rhel7 & .ci-operator.yaml are up to date (they are. [Thanks, ART team](https://github.com/openshift/vertical-pod-autoscaler-operator/pull/163)!)
* Update manifests to match upstream
* Updated ci + release images to `FROM registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.22-openshift-4.17` (`images/ci/Dockerfile`,`images/operator-registry/Dockerfile.registry.ci`, `Dockerfile`, `Dockerfile.rhel`)
* Updated ci base image from `openshift/origin-v4.0` to `registry.ci.openshift.org/ocp/4.17:base-rhel9`

Had to update some struct initialization and func arguments because of a `controller-runtime` upgrade to v18.4 from v17.0 https://github.com/kubernetes-sigs/controller-runtime/pull/2783

### CODE REVIEW

**Comment by joelsmith** [comment]
As discussed on Slack, `sed -i 's/4.16/4.17/g' $(git grep -l 4.15 manifests/)` should have been `sed -i 's/4.16/4.17/g' $(git grep -l 4.16 manifests/)`

Nice work on the controller runtime related updates!

**Comment by maxcao13** [comment]
I was looking at it, should `golint` and its scripts + ci yamls still exist? It seems to be deprecated in favour of `go vet` and Staticcheck https://github.com/golang/lint

**Comment by maxcao13** [comment]
Seems like `registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.22-openshift-4.17` doesn't include `jq` and that causes `manifest-diff*.sh` scripts to fail.

**Comment by maxcao13** [comment]
Updated with new release images.

File: hack/manifest-diff.sh [hunk 1]
```diff
@@ -75,9 +75,10 @@ if [ "$NO_DOCKER" = "1" -o -n "$IS_CONTAINER" ]; then
...
 else
   podman run --rm \
+    -it \
     --env IS_CONTAINER=TRUE \
     --volume "${repo_base}:/go/src/github.com/openshift/${repo_name}:z" \
     --workdir "/go/src/github.com/openshift/${repo_name}" \
-    registry.ci.openshift.org/openshift/release:golang-1.18 \
-    ./hack/manifest-diff.sh "${@}"
+    registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.22-openshift-4.17 \
+    /bin/bash
```
Comment (joelsmith) [inline hack/manifest-diff.sh:83]: It looks like this was left unfinished. Should it still be `./hack/manifest-diff.sh "${@}"`?

**Review by joelsmith — CHANGES_REQUESTED** [review]
I only see one issue with the manifest diff test. Everything else looks great! I'll watch for that to get updated then I'll add the LGTM label.

/approve

**Review by joelsmith — APPROVED** [review]
/lgtm

**Pull request merged**